nvtabular/io/parquet.py

```diff
@@ -661,14 +685,63 @@ def _write_thread(self):
             finally:
                 self.queue.task_done()
 
+    @classmethod
+    def write_special_metadata(cls, md, fs, out_dir):
+        """Write global _metadata file"""
+        raise (NotImplementedError)
+
+    def _close_writers(self):
+        """Close writers and return extracted metadata"""
+        raise (NotImplementedError)
+
+    def _bytesio_to_disk(self):
+        md = {}
+        for bio, path in zip(self.data_bios, self.data_paths):
+            df = self._read_parquet(bio)
+            bio.close()
+            if self.shuffle == Shuffle.PER_WORKER:
+                df = _shuffle_df(df)
+            md[path] = self._to_parquet(df, path)
+        return md
```

Comment:
Notice that we cannot use the original metadata for the the data written to `bio`, because the row-groups may be different when we re-write the whole file in one go.